R/backends.R

```diff
@@ -24,6 +24,9 @@ parse_model <- function(model, backend, ...) {
 .parse_model_cmdstanr <- function(model, silent = 1, ...) {
   require_package("cmdstanr")
   temp_file <- cmdstanr::write_stan_file(model)
+  if (cmdstanr::cmdstan_version() >= "2.29.0") {
+    .canonicalize_stan_model(temp_file, overwrite_file = TRUE)
+  }
   out <- eval_silent(
     cmdstanr::cmdstan_model(temp_file, compile = FALSE, ...),
     type = "message", try = TRUE, silent = silent
```

Comment:
@paul-buerkner I might be wrong, but I think if this function just returned `model`, that might work as the model is no canonicalized? 
I honestly don't know what `bridgesampling` uses (where it gets the code from), but the idea to fix this would be to only canonicalize right before calling `cmdstan_model()`.

Reply:
Here is a reprex:

```R
fit3 <- brm(
  time | cens(censored) ~ age * sex + disease + (1|patient),
  data = kidney, family = lognormal(), 
  backend = "cmdstanr"
)

bridge_sampler(fit3)
```

The problem is that "brmsfit$model" also has the new array syntax but `add_rstan_model`, which is called for bridgesampling and friends doesn't know that. I have just fixed it on master.